src/pages/Lend/LoanChannelCategoryPage.vue

```diff
@@ -0,0 +1,293 @@
+		}
+	},
+};
+
+// Turn an object of graphql variables into an object of url query parameters
+function toUrlParams(variables) {
+	return _mapValues(urlParamTransform, ({ to }) => to(variables));
+}
+
+// Turn an object of url query parameters into an object of graphql variables
+function fromUrlParams(params) {
+	return _merge({}, ..._invokeMap(urlParamTransform, 'from', params));
+}
+
+function getTargetedChannel(targetedRoute, allChannels) {
+	const loanChannels = _get(allChannels, 'lend.loanChannels.values');
+	// map id from loan channels
+	const targetedLoanChannel = _filter(
+		loanChannels,
+		loanChannel => {
+			return loanChannel.url.indexOf(targetedRoute) !== -1;
+		}
+	);
+	// isolate targeted loan channel id
+	return _get(targetedLoanChannel[0], 'id') || null;
+}
+
+export default {
+	components: {
+		WwwPage,
+		GridLoanCard,
+		KvPagination,
+		LoadingOverlay,
+		ViewToggle
+	},
+	inject: ['apollo'],
+	metaInfo: {
+		title: 'Fundraising loans'
+	},
+	data() {
+		return {
+			offset: 0,
+			limit: loansPerPage,
+			filters: { },
+			targetedLoanChannelID: null,
+			loanChannel: () => {},
+			isVisitor: true,
+			itemsInBasket: [],
+			pageQuery: { page: '1' },
+			loading: false,
+		};
+	},
+	computed: {
+		urlParams() {
+			return toUrlParams({
+				offset: this.offset,
+			});
+		},
+		loans() {
+			return _get(this.loanChannel, 'loans.values') || [];
+		},
+		totalCount() {
+			return _get(this.loanChannel, 'loans.totalCount') || 0;
+		},
+		loanQueryVars() {
+			return {
+				ids: [this.targetedLoanChannelID],
+				limit: this.limit,
+				offset: this.offset
+			};
+		},
+		filterUrl() {
+			// initial release sends us back to /lend
+			return `/lend/${this.$route.params.category || ''}`;
+		}
+	},
+	apollo: {
+		preFetch(config, client, args) {
+			return client.query({
+				query: loanChannelPageQuery
+			}).then(({ data }) => {
+				// filter routes on route.param.category to get current path
+				const targetedLoanChannelURL = _get(args, 'route.params.category');
+				// isolate targeted loan channel id
+				const targetedLoanChannelID = getTargetedChannel(targetedLoanChannelURL, data);
+				// extract query
+				const pageQuery = _get(args, 'route.query');
+				// query our targeted loan channel
+				return client.query({
+					query: loanChannelQuery,
+					variables: _merge(
+						{
+							ids: [targetedLoanChannelID],
+							limit: loansPerPage
+						},
+						fromUrlParams(pageQuery)
+					)
+				});
+			}).then(() => {
+				// TODO: REMOVE Once Lend Increment Button EXP ENDS
+				// Pre-fetch the setting for lend increment button
+				return client.query({ query: experimentSetting, variables: { key: 'uiexp.lend_increment_button' } });
+			}).then(() => {
+				// TODO: REMOVE Once Lend Increment Button EXP ENDS
+				// Pre-fetch the assigned version for lend increment button
+				return client.query({ query: experimentQuery, variables: { id: 'lend_increment_button' } });
+			});
```

Comment:
Since these experiment queries don't depend on the results of the other queries at all, you could actually run them in parallel with the first query by using Promise.all

Reply:
ooh good idea!